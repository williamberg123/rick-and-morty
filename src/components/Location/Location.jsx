import React from 'react';

import PropTypes from 'prop-types';

import './Location.css';

export default function Location({ name, type, dimension }){
    return (
        <div className="Location">
            <div className="Location-name">{name}</div>
            <div className="Location-hidden-info">
                <div className="Location-type">Tipo: 
                    <span>
                        {type === 'unknown'
                            ? 'Desconhecido'
                            : type
                        }
                    </span>
                </div>
                <div className="Location-dimension">
                    Dimens√£o:
                    <span>
                        {dimension === 'unknown'
                            ? 'Desconhecida'
                            : dimension
                        }
                    </span>
                </div>
            </div>
        </div>
    );
}

Location.propTypes = {
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    dimension: PropTypes.string.isRequired
};