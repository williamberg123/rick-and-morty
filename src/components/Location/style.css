.Location {
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: white;
    color: black;
    font-weight: bold;
    padding: 5px;
    border-radius: 10px;
    border: 1px solid black;
    box-shadow: 3px 3px 10px rgb(138, 138, 138);
    transition: 0.5s;
}

.Location:hover {
    padding: 10px;
    background-color: rgb(8, 65, 104);
}

.Location:hover .Location-type {
    color: white;
}
.Location:hover .Location-hidden-info {
    display: flex;
}

.Location-name {
    color: red;
}

.Location-hidden-info {
    display: none;
    flex-direction: column;
    transition: 0.5s;
    color: white;
}

.Location-hidden-info span {
    color: red;
}

@media (max-width: 450px) {
    .Location-hidden-info {
        font-size: 0.7rem;
    }
}