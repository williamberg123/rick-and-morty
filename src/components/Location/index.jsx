import React, { memo } from 'react';

import PropTypes from 'prop-types';

import './style.css';

function Location({ name, type, dimension }){
    return (
        <div className="Location">
            <div className="Location-name">{name}</div>
            <div className="Location-hidden-info">
                <div className="Location-type">Tipo: 
                    <span>
                        {type === 'unknown'
                            ? 'Desconhecido'
                            : type
                        }
                    </span>
                </div>
                <div className="Location-dimension">
                    Dimens√£o:
                    <span>
                        {dimension === 'unknown'
                            ? 'Desconhecida'
                            : dimension
                        }
                    </span>
                </div>
            </div>
        </div>
    );
}

Location.propTypes = {
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    dimension: PropTypes.string.isRequired
};

export default memo(Location);