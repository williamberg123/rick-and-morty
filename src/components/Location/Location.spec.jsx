import React from 'react';
import { render, screen } from '@testing-library/react';
import Location from './index';

describe('<Location />', () => {
    it('should render corretcly', () => {
        const { debug } = render(<Location name="Earth" type="planet" dimension="C-137" />);

        expect(screen.getByText('Earth')).toBeInTheDocument();
        expect(screen.getByText(/tipo:/i)).toBeInTheDocument();
        expect(screen.getByText(/DimensÃ£o:/i)).toBeInTheDocument();
        // debug();
    });

    it('should to have text desconhecido when type is unknown', () => {
        render(<Location name="Earth" type="unknown" dimension="C-137" />);
        
        expect(screen.getByText(/desconhecido/i)).toBeInTheDocument();
    });

    it('should to have text desconhecida when dimension is unknown', () => {
        const { debug } = render(<Location name="Earth" type="planet" dimension="unknown" />);
        
        expect(screen.getByText(/desconhecida/i)).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        render(<Location name="Earth" type="planet" dimension="C-137" />);
        expect(screen.getByText('Earth').parentElement)
            .toMatchSnapshot();
    });
});

// Testado