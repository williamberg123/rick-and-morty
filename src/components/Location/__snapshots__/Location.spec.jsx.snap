// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Location /> should match snapshot 1`] = `
<div
  class="Location"
>
  <div
    class="Location-name"
  >
    Earth
  </div>
  <div
    class="Location-hidden-info"
  >
    <div
      class="Location-type"
    >
      Tipo:
      <span>
        planet
      </span>
    </div>
    <div
      class="Location-dimension"
    >
      Dimens√£o:
      <span>
        C-137
      </span>
    </div>
  </div>
</div>
`;
